keras_nlp/utils/text_generation.py

```diff
@@ -0,0 +1,113 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Text generation utilities."""
+
+import tensorflow as tf
+
+
+def _handle_end_token(next_token, end_token_received, end_token_id):
+    filtered_next_token = next_token * (1 - end_token_received)
+    end_token_received = tf.cast(
+        filtered_next_token == end_token_id, dtype=next_token.dtype
+    )
+    return filtered_next_token, end_token_received
+
+
+def generate_text_greedy(
+    get_next_token_probability_fn, seed, max_length, end_token_id=None
+):
+    """
+    Text generation utility based on greedy search.
+
+    Greedy search always appends the token having the largest probability to
+    existing sequence.
+
+    Args:
+        get_next_token_probability_fn: a callable, which takes in input_sequence
+            and output the probability distribution of the next token.
+        seed: a list, the initial tokens to append generated tokens.
```

Comment:
This is my bad, but I'm realizing seed would conflict with a rng seed to make the output deterministic. We should probably call this `input_ids` or `starting_ids`

Reply:
I will go with `input_ids`